src/system-update-generator/system-update-generator.c

```diff
@@ -18,19 +19,22 @@
 static const char *arg_dest = NULL;
 
 static int generate_symlink(void) {
-        const char *p = NULL;
+        _cleanup_free_ char *j = NULL;
 
-        if (laccess("/system-update", F_OK) < 0) {
-                if (errno == ENOENT)
-                        return 0;
+        FOREACH_STRING(p, "/system-update", "/etc/system-update") {
+                if (laccess(p, F_OK) >= 0)
+                        goto link_found;
 
-                log_error_errno(errno, "Failed to check for system update: %m");
-                return -EINVAL;
+                if (errno != ENOENT)
+                        log_error_errno(errno, "Failed check for %s symlink, ignoring: %m", p);
```

Comment:
Please use log_warning_errno(), as the failure is not critical now.

Also, "Failed to check if %s is symlink".

Reply:
I'll slightly reword to "Failed to check if %s symlink exists", if that's ok? That's more in the spirit of what we are trying to do here.

Reply:
Repushed